### ISSUE TITLE
SenML converter

### ISSUE THREAD

**Issue by almul0** [koestler/go-mqtt-to-influx#2]
Hi there, I am developing a senml converter. Now I am stuck at the test stage to make a pull request, I think it could be a nice feature. I am not fluent with Go and I am not used to implementing TDD. Could you please send me instructions about how to test a converter?

Thanks in advance,
Alberto

**Comment by koestler** [comment]
Hi there!

I've never worked with the senml standard and would have to look into that. But is it also sending some kind of messages over MQTT which you now want to redirect to an influx-db?

If so, i could probably help to develop a converter. Or do you already have any code for that?

What we basically would need is a few example messages of as many different sensors as possible. Than we could figure out how this data is best stored in the database and than write/test the converter.

Best regards,
Lorenz

**Comment by almul0** [comment]
Hi Lorentz:

Yes, although it is in the proposed status right now, we found it very convenient as a multipurpose information exchange standard. I already develop a little converter based on go-ve-sensor.go code and using this implementation (https://github.com/mainflux/senml) for message normalization. 

In my converter, I extract the measurement field from the topic, tags from unit and base name and fields and values from name and value respectively. One data point from each resolved measure.

I tried to develop a test with the second message defined in https://tools.ietf.org/html/rfc8428#section-5.1.2, but as it has different timestamps I got stuck.  Also, I  was not able to run any kind of tests nor generate mocks.

I already push a new branch https://github.com/howlab/go-mqtt-to-influx/tree/senml-converter

Thanks in advance,
Alberto
